src/execution/index/art/node.cpp

```diff
@@ -186,4 +188,208 @@ void Node::Erase(Node *&node, idx_t pos, ART &art) {
 	}
 }
 
+void Node::ResolvePrefixesAndMerge(ART *l_art, ART *r_art, Node *&l_node, Node *&r_node, idx_t depth,
+                                   Node *&l_node_parent, idx_t l_node_pos, Node *&r_node_parent, idx_t r_node_pos) {
+
+	if (!l_node) {
```

Comment:
This check can probably be moved to the parent instead of staying in the recursion.